Content.Shared/Weapons/Ranged/Components/BallisticAmmoProviderComponent.cs

```diff
@@ -48,6 +48,22 @@ public sealed partial class BallisticAmmoProviderComponent : Component
     [AutoNetworkedField]
     public bool Cycleable = true;
 
+    /// <summary>
+    /// Is the firearm currently cycled?
+    /// It cannot fire if is it not cycled.
+    /// Must be manually cycled if it is not cycled.
+    /// </summary>
+    [ViewVariables(VVAccess.ReadWrite), DataField("cycled")]
+    [AutoNetworkedField]
+    public bool Cycled = true;
+
+    /// <summary>
+    /// Automatically cycles the firearm after firing a round
+    /// </summary>
+    [ViewVariables(VVAccess.ReadWrite), DataField("autoCycle")]
+    [AutoNetworkedField]
+    public bool AutoCycle = true;
+
```

Comment:
Just make it use chambermagazines if you want the functionality back.

Reply:
What do I do with the chambermagazines? The ChamberMagazineAmmoProvider component requires a magazine for it to function, and none of these guns have magazines.

Reply:
I'm having second thoughts on it so let me think about it a bit and I'll come back to you.

Reply:
Have you decided anything yet?

Reply:
Make cycled nullable instead and update it, then add a method for IsCycled that checks if it's null or true.

Reply:
Like this? https://github.com/space-wizards/space-station-14/pull/19287/commits/bb6266dd442e81c0a8918aae5aa5d8e671d75312